7_react/react1/week2/TODO-LIST/src/App.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import toDoItems from "./component/TodoItems";
+import Timer from "./component/timer";
+import "./App.css";
+
+
+function App() {
+  const [toDoItem, setToDoItems] = useState(toDoItems);
+// Function to add a random item
+  const addRandomItem = () => {
+    const randomId = Math.floor(Math.random() * 1000);
+    const newItem = {
+      id: randomId,
+      description: `Task ${randomId}`,
+      deadline: "Some deadline", 
+      done: false // Initially task is not done
+    };
+    setToDoItems([...toDoItems, newItem]);
+  };
+
+  // mark as done
+  const markAsDone = (id) => {
+    setToDoItems(
+      toDoItems.map((item) =>
+        item.id === id ? { ...item, done: true } : item
+      )
+    );
+  };
+  //  mark as not done
+  const markAsNotDone = (id) => {
+    setToDoItems(
+      toDoItems.map((item) =>
+        item.id === id ? { ...item, done: false } : item
+      )
+    );
+  };
+  // Function to delete an item
+  const deleteItem = (id) => {
+    setToDoItems(toDoItems.filter((item) => item.id !== id));
+  };
+
+  return (
+    <div className="container">
+      <h2>Todo List</h2>
+      <Timer />
+      <button onClick={addRandomItem}>Add Items Here</button>
+      {toDoItems.length === 0 ? (
+        <p>No items</p>
+      ) : (
+      <ul>
+        {toDoItems.map((todos,index) => (
```

Comment:
I will note one last time why it is important for you to make the first comment's correction. In your code currently `toDoItems` is the constant array that you import. And in this call, you are rendering the elements of that array. However, this array never changes, since it is not the state variable! This is why no matter what you click in the current version of the code, the state stays the same. That must have been frustrating. 😄 

Again, just by making the first comment's correction, this will also work fine.

P.S: You do not actually use the array `index` for anything, so you can just remove that parameter from the callback of `map()`. Moreover, since you are iterating over every individual todo, the other parameter should rather be called `todo`, instead of `todos`. 😉